### PR TITLE
ex02-module-class

### PR DESCRIPTION
**Pull request by hubertkozik**
Second exercise.

### CODE REVIEW

File: exercises/02-module-class/color-generator-2.0.mjs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class ColorsGenerator {
+
+    colors = ['red', 'green', 'blue'];
```
Comment (pawelfus) [inline exercises/02-module-class/color-generator-2.0.mjs]: This field should be [private](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields)

From [exercise description](https://github.com/hubertkozik/highcharts-modules/tree/main/exercises/02-module-class):
> Implement a [Class](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) with:
> 
> a) Public methods:
> 
> `get(n?: number): Array<string>;`
> `reset(): void;`
**> b) Private properties and methods:**
> 
> `colors: Array<string>`
> any other properties and methods you need

File: exercises/02-module-class/color-generator-2.0.mjs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    i = 0;
+
+    constructor() {
+    }
+
+    get(n) {
```
Comment (pawelfus) [inline exercises/02-module-class/color-generator-2.0.mjs]: Not a bug, but during your code reviews in this classroom, if possible please promote [default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters) usage

I can't complain about your solution, it is smart :D But junior devs tend to solve it in different ways, for example: 
```ts
get(n) {
  const count = n || 1;
}
```
Or:
```ts
get(n) {
  if (!n) {

  }  else ...
}
```

Which could be simpler with default params

File: exercises/02-module-class/color-generator-2.0.mjs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class ColorsGenerator {
+
+    colors = ['red', 'green', 'blue'];
+    i = 0;
```
Comment (pawelfus) [inline exercises/02-module-class/color-generator-2.0.mjs]: Not a bug per se, but a bit more descriptive name would be nice 😅 For example `counter` or `index`.

**Review by pawelfus — CHANGES_REQUESTED** [review]

File: exercises/02-module-class/color-generator-2.0.mjs (hunk 2 above)
Comment (hubertkozik) [reply exercises/02-module-class/color-generator-2.0.mjs]: I have changed this part of the code to use default parameters, but in my opinion, the code looked better in my first solution 😀 But I understand, that default parameters should be promoted and I will remember about it, when doing some future code reviews 🙂

**Pull request closed without merging**